services/core-api/app/api/mines/permits/permit_conditions/models/permit_conditions.py

```diff
@@ -54,6 +53,9 @@ def sub_conditions(self):
 
     @hybrid_property
     def step(self):
+        if self._step:
+            return self._step
```

Comment:
Added in a `_step` column that contains the `step` extracted for the permit, and return that if defined instead of the auto-generated step. Note: This will require some work to make work nicely with auto-generation of step when adding / editing conditions